crates/rpc-types-engine/src/envelope.rs

```diff
@@ -0,0 +1,194 @@
+//! Optimism execution payload envelope in network format and related types.
+//!
+//! This module uses the `snappy` compression algorithm to decompress the payload.
+//! The license for snappy can be found in the `SNAPPY-LICENSE` at the root of the repository.
+
+use alloy_primitives::{keccak256, Signature, B256};
+use alloy_rpc_types_engine::ExecutionPayload;
+use derive_more::derive::{Display, From};
+
+/// Optimism execution payload envelope in network format.
+///
+/// This struct is used to represent payloads that are sent over the Optimism
+/// CL p2p network in a snappy-compressed format.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct OptimismNetworkPayloadEnvelope {
```

Comment:
```suggestion
pub struct OpNetworkPayloadEnvelope {
```
recent new naming convention